analytics/src/main/java/com/segment/analytics/ConnectionFactory.java

```diff
@@ -79,7 +81,15 @@ public HttpURLConnection attribution(String writeKey) throws IOException {
    * #attribution(String)} and {@link #projectSettings(String)}.
    */
   protected HttpURLConnection openConnection(String url) throws IOException {
-    HttpURLConnection connection = (HttpURLConnection) new URL(url).openConnection();
+    URL requestedURL;
+
+    try {
+      requestedURL = new URL(url);
+    } catch (MalformedURLException e) {
+      throw new IOException("Attempted to use malformed url: " + url);
```

Comment:
we should propagate the underlying error here.

Reply:
Thanks!  I didn't realize I could include multiple exceptions in the signature.